Source/WebCore/inspector/agents/InspectorDOMStorageAgent.cpp

```diff
@@ -214,7 +214,7 @@ RefPtr<StorageArea> InspectorDOMStorageAgent::findStorageArea(Protocol::ErrorStr
     }
 
     if (!*isLocalStorage)
-        return m_inspectedPage.sessionStorage()->storageArea(targetFrame->document()->securityOrigin());
+        return m_inspectedPage.sessionStorage(targetFrame->document()->topOrigin())->storageArea(targetFrame->document()->securityOrigin());
```

Comment:
It may be worth storing auto& document = *targetFrame->document(); for this line and the one below.

Reply:
Added.